geoapps/drivers/components/locations.py

```diff
@@ -110,6 +111,12 @@ def get_locations(self, uid: UUID) -> np.ndarray:
             msg += " Object type should be Grid2D or point-like."
             raise (ValueError(msg))
 
+        if data_object.uid == self.params.topography_object:
+            if self.params.topography is not None:
+                elev = self.workspace.get_entity(self.params.topography)[0].values
+                if not np.all(locs[:, 2] == elev):
+                    locs[:, 2] = elev
+
```

Comment:
I am not following here. Why over-writing the locs Z on topo? Looks like a repeat from the initialize of the InversionTopography class

Reply:
Looks like I meant to remove one of them.  I opted to extend the get_locations method on the topography class to handle this, and removed from the locations.